src/memory-analyzer/analyze_symbol.cpp

```diff
@@ -330,13 +331,18 @@ exprt gdb_value_extractort::get_pointer_to_member_value(
     if(maybe_member_expr.has_value())
       return *maybe_member_expr;
     UNREACHABLE;
+    return index_exprt{
+      struct_symbol->symbol_expr(),
+      from_integer(
+        member_offset / get_type_size(expr.type().subtype()), index_type())};
   }
 
   const auto it = values.find(memory_location);
   // if the structure we are pointing to does not exists we need to build a
   // temporary object for it: get the type from symbol table, query gdb for
   // value, allocate new object for it and then store into assignments
   if(it == values.end())
+  if(struct_symbol->symbol_expr().type().id() == ID_pointer)
```

Comment:
This looks very weird?!

Reply:
What about `struct_symbol->type.id()` ? Changed.